client/state/sharing/services/selectors.js

```diff
@@ -9,6 +19,58 @@ export function getKeyringServices( state ) {
 }
 
 /**
+ * Returns an object of service objects with the specified type.
+ *
+ * @param  {Object} state Global state tree
+ * @param  {String} type  Type of service. 'publicize' or 'other'.
+ * @return {Array}        Keyring services, if known.
+ */
+export function getKeyringServicesByType( state, type ) {
+	return filter( getKeyringServices( state ), { type: type } );
+}
+
+/**
+ * Returns an object of eligible service objects with the specified type.
+ *
+ * @param  {Object} state Global state tree
+ * @param  {Object} site  Site object.
+ * @param  {String} type  Type of service. 'publicize' or 'other'.
+ * @return {Array}        Keyring services, if known.
+ */
+export function getEligibleKeyringServices( state, site, type ) {
+	const services = getKeyringServicesByType( state, type );
+
+	if ( ! site ) {
+		return services;
+	}
+
+	return services.filter( ( service ) => {
+		// Omit if the site is Jetpack and service doesn't support Jetpack
+		if ( site.jetpack && ! service.jetpack_support ) {
+			return false;
+		}
+
+		// Omit if Jetpack module not activated
+		if ( site.jetpack && service.jetpack_module_required &&
+			! site.isModuleActive( service.jetpack_module_required ) ) {
+			return false;
+		}
+
+		// Omit if service is settings-oriented and user cannot manage
+		if ( 'eventbrite' === service.ID && ! utils.userCan( 'manage_options', site ) ) {
```

Comment:
We should use the [`canCurrentUser`](https://github.com/Automattic/wp-calypso/blob/08b32a64728927babb7b093ba988e695c37eaafd/client/state/current-user/selectors.js#L91-L110) selector here instead. `lib/site/utils` should be deprecated in favor of selectors. Also easier to call, only needing a site ID instead of the full site object.